Seated various misplaced methods

jawa9000 opened the pull request:
https://jira.appcelerator.org/browse/TIDOC-2489

The following methods were not properly seated in the YAML heirarchy:Point, Gradient, GradientColorRef, Dimension, and AnimationOption. With these methods seated under the properties tree, the local build properly shows these methods as well as the entire document.

jawa9000 committed:
Re-applied the livePhoto contents

Per discussion with Hans and Fokke, I re-applied the content I removed
in relation to livePhoto.

jawa9000 committed:
Merge remote-tracking branch 'appcelerator/master'

jawa9000 committed:
Fixed improperly located methods (TIDOC-2489)

The following methods were not properly seated in the YAML heirarchy:
Point, Gradient, GradientColorRef, Dimension, and AnimationOption.

hansemannn commented:
Hey Brian, we have some merge conflicts with latest master here, can you check?

hansemannn commented on apidoc/Titanium/Media/Media.yml:
```diff
@@ -1526,7 +1526,9 @@ properties:
-        Array of media type constants to allow.
+        Array of media type constants to allow. If you want to allow live photos with
+        Titanium.Media.MEDIA_TYPE_LIVEPHOTO on iOS 9.1 and later, you also need to specify
+        at least Titanium.Media.MEDIA_TYPE_PHOTO as a fallback.
```
Incorrect, because the camera doesn't support live photos (yet). That's why I didn't add it there :-)

hansemannn commented on apidoc/Titanium/UI/iOS/LivePhotoView.yml:
```diff
@@ -60,6 +60,14 @@ events:
         type: Number
 
+properties:
+  - name: livePhoto
```
Not sure why this doesn't throw merge conflicts, since the master already contains [this property](https://github.com/appcelerator/titanium_mobile/blob/master/apidoc/Titanium/UI/iOS/LivePhotoView.yml#L22). Can you check? Thanks man!

jawa9000 committed:
fix for TIDOC-2489

The following methods were not properly seated in the YAML heirarchy:
Point, Gradient, GradientColorRef, Dimension, and AnimationOption.

jawa9000 committed:
Resolving conflicts

jawa9000 committed:
Fixed links mentioned in TIDOC-2487

hansemannn commented on apidoc/Titanium/UI/iPhone/AlertDialogStyle.yml:
```diff
@@ -9,7 +9,7 @@ platforms: [iphone, ipad]
 deprecated:
     since: "6.0.0"
-    notes: Use [Titanium.UI.iOS.AlertDialogStyle](Titanium.UI.iOS.AlertDialogStyle) instead.
+    notes: Use Titanium.UI.iOS.AlertDialogStyle instead.
```
Why not use `<Ti.UI.iOS.AlertDialogStyle>` which should auto-link it?

jawa9000 replied:
We are having problems crossing linking if people use the short-hand of Titanium (Ti). Right now, our document system doesn't treat Titanium.whatever and Ti.whatever the same. Some links are broken, in particular those using the incorrect module prefix. Ultimately, we need to unify which one we are going to use.

hansemannn replied:
Makes sense! But still, wouldn't the Autolinking make sense (using <>)?

hansemannn commented on apidoc/Titanium/UI/iPhone/iPhone.yml:
```diff
@@ -181,7 +181,7 @@ properties:
     deprecated:
         since: "6.0.0"
-        notes: Use [Ti.UI.iOS.appBadge](Ti.UI.iOS.appBadge) instead.
+        notes: Use Titanium.UI.iOS.appBadge instead.
```
Same as above

jawa9000 committed:
Resulting conflict

jawa9000 committed:
resolving conflict and fixing properties

Found four properties that were sitting outside the properties tree:
hideStatusBarParams, showStatusBarParams, animated, and animationStyle.

jawa9000 committed:
update from master

jawa9000 committed:
update from appcelerator/master

hansemannn commented on apidoc/Titanium/UI/iPhone/iPhone.yml:
```diff
@@ -216,42 +216,26 @@ properties:
-        <Titanium.UI.iPhone.StatusBar.ANIMATION_STYLE_SLIDE>, 
-        or <Titanium.UI.iPhone.StatusBar.ANIMATION_STYLE_FADE>.
-    type: Number
+  - name: hideStatusBarParams
```
Why do we do this? The `----` is meant to be used  for dictionaries that need to be passed to other API's. They are no properties / methods.

hansemannn commented on apidoc/Titanium/UI/ListView.yml:
```diff
@@ -851,7 +851,11 @@ properties:
         since: "5.2.0"
+<<<<<<< HEAD
+        notes: Use tableSeparatorInsets instead
+=======
```
Merge problems

hansemannn commented:
Hey @jawa9000, there are lots of merge issues resulting on diff-ing the master. Can you submit a new one that includes the changes you focussed on initially?
